File: works/tmp/11060.kt
import java.util.*
import java.io.*;
import java.io.BufferedWriter;
import java.io.BufferedReader;
import java.io.InputStreamReader;
import java.io.OutputStreamWriter;

fun main()
{
    val br = BufferedReader(InputStreamReader(System.`in`))

    var n = br.readLine().toInt()
    var tmp = br.readLine().split(' ')
    var a = Array<Int>(n+1){0}
    var d = Array<Int>(n+1){0}
    for(i in 0 until n) a[i] = tmp[i].toInt()

    if(n == 1) {print(0); return}

    d[0] = 0
    for(i in 1 until n)
    {
        var lowest = 2000000000
        var trigger = false
        for(j in 0 until i)
        {
            if(j + a[j] >= i) { lowest = Math.min(lowest, d[j]+1); trigger = true}
        }
        if(!trigger) { print(-1); return }
        d[i] = lowest
    }
    print(d[n-1])
}

File: works/tmp/11060.cpp
#include <iostream>
#include <algorithm>
using namespace std;


int main()
{
    ios::sync_with_stdio(false); cin.tie(NULL);

    int a[1010] = {0,}, d[1010] = {0,};
    int n; cin >> n;
    for(int i = 0; i < n; i++) cin >> a[i];
    if(n == 1) {cout << 0; return 0;}

    d[0] = 0;
    for(int i = 1; i < n; i++)
    {
        int lowest = 2000000000;
        bool trigger = false;
        for(int j = 0; j < i; j++)
        {
            if(j + a[j] >= i) { lowest = min(lowest, d[j]+1); trigger = true; }
        }
        if(!trigger) { cout << -1; return 0; }
        d[i] = lowest;
    }

//    for(int i = 0; i < n; i++) cout << d[i] << ' '; cout << endl;

//    int answer = d[n-1] == 0 ? -1 : d[n-1];
    cout << d[n-1];
}